Sources/FoundationEssentials/Calendar/Calendar_Recurrence.swift

```diff
@@ -346,6 +379,11 @@ extension Calendar {
                 if weekAction == .expand {
                     // In a yearly recurrence with weeks specified, results do not land on any specific month
                     componentCombinations.weeksOfYear = recurrence.weeks
+                    if componentCombinations.weekdays == nil {
+                        if let weekdayIdx = components.weekday, let weekday = Locale.Weekday(weekdayIdx) {
+                            componentCombinations.weekdays = [.every(weekday)]
```

Comment:
In adding tests, I found another regression from #981: if we are expanding weeks, we should always have a weekday set. The test case we had (`testYearlyRecurrenceWithWeekNumberExpansion`) just happened to work without the fix since the anchor was the very start of the search range.

Reply:
Actually no, I was misled -- we already handle this above. I just went down the wrong rabbit hole while writing test cases.